Modules/Keychain/Sources/KeychainError.swift

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2024 Jamit Labs GmbH. All rights reserved.
+
+import Foundation
+
+enum KeychainError: Error, LocalizedError {
+    case itemNotFound
+    case accessError(status: String)
+    case decodingError(error: Error)
+    case encodingError(error: Error)
+    case saveItemToKeychain(status: String)
+    case deleteItem
+
+    var errorDescription: String? {
+        switch self {
+        case .itemNotFound: return "The requested item could not be found in the keychain"
+        case .accessError: return "The keychain item could not be accessed"
+        case .decodingError: return "The retrieved keychain data could not be decoded"
+        case .encodingError: return "The item could not be encoded"
+        case .saveItemToKeychain: return "The item could not be saved to the keychain"
+        case .deleteItem: return "The item could not be removed from the keychain"
+        }
+    }
+
+    var failureReason: String? {
+        switch self {
+        case let .accessError(status: error): return error
+        case let .decodingError(error: error): return error.localizedDescription
+        case let .encodingError(error: error): return error.localizedDescription
+        case let .saveItemToKeychain(status: error): return error
+        default: return nil
+        }
+    }
+
+    var recoverySuggestion: String? {
+        switch self {
+        case .itemNotFound: return "Please make sure that the item exists in the keychain before you try to access it"
+        case .deleteItem: return "Please make sure that the item exists in the keychain before you try to delete it"
```

Comment:
Is this really an error case? In my opinion, we should see the scenario of trying to delete an item which is not present as an info rather than an error.

What do you think?

Reply:
This error seems to be thrown only if the deletion was unsuccessful, but no if the error was "item not found", so this recovery suggestion is in my opinion misleading.
The error is only thrown if a different error occurs during deletion, so I am not sure what recovery suggestion we could give here. For now, I deleted the switch case to remove the misleading recovery suggestion.